core/Pos/Binary/Class/Store.hs

```diff
@@ -0,0 +1,259 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DeriveFunctor       #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | Serialization-related types
+
+module Pos.Binary.Class.Store
+       (
+         putConst
+       , putField
+       , convertToSizeNPut
+       -- * Store re-exports
+       , Size(..)
+       , Peek
+       , Poke
+
+       -- * Poke with size of the poke.
+       , PokeWithSize (..)
+       , putS
+
+       -- * The 'StaticSize' wrapper
+       , StaticSize(..)
+
+       -- * Primitives for limiting serialization
+       , isEmptyPeek
+       , lookAhead
+       , mkPoke
+       , execPoke
+
+       , convertSize
+       , combineSize
+       , sizeOf
+       , sizeAddField
+       ) where
+
+import           Universum
+
+import           Data.Functor.Contravariant (contramap)
+import           Data.Store                 (Size (..))
+import           Data.Store.Core            (Peek (..), PeekResult (..), Poke (..))
+import qualified Data.Store.Core            as Store
+import           Data.Store.Internal        (PeekException (..), StaticSize (..))
+import qualified Data.Store.Internal        as Store
+
+import           Pos.Binary.Class.Core      (Bi (..), getSize)
+
+----------------------------------------------------------------------------
+-- Helpers to implement @sizeNPut@
+----------------------------------------------------------------------------
+
+-- [CSL-1122] shouldn't this mappend results as well? like Monoid (IO a) does
+instance Monoid a => Monoid (Poke a) where
```

Comment:
mda, seems it make sense define `Monoid (Poke ())`, because `Monoid (Poke a)` isn't needed at all